### PR TITLE
feat: ui components migration

### PR DESCRIPTION
**Pull request by Rickk137**
https://sovryn.monday.com/boards/2218344956/pulses/2939205582
https://sovryn.monday.com/boards/2218344956/pulses/2693964725

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
```
Comment (tiltom) [inline .prettierrc]: Why did you add this file? We have our configs in packages and we use configuration from `eslint-config-custom`.

File: .vscode/extensions.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{
```
Comment (tiltom) [inline .vscode/extensions.json]: This doesn't belong to the repo, it's probably just some local config for you. Don't commit it.

File: .vscode/launch.json [hunk 3]
```diff
@@ -0,0 +1,15 @@
+{
```
Comment (tiltom) [inline .vscode/launch.json]: This doesn't belong to the repo, it's probably just some local config for you. Don't commit it.

File: .vscode/postcss-custom-data.json [hunk 4]
```diff
@@ -0,0 +1,29 @@
+{
```
Comment (tiltom) [inline .vscode/postcss-custom-data.json]: This doesn't belong to the repo, it's probably just some local config for you. Don't commit it.

File: .vscode/settings.json [hunk 5]
```diff
@@ -0,0 +1,43 @@
+{
```
Comment (tiltom) [inline .vscode/settings.json]: This doesn't belong to the repo, it's probably just some local config for you. Don't commit it.

File: packages/ui/.storybook/preview.js [hunk 6]
```diff
@@ -1,15 +1,26 @@
...
+import "../styles/index.css";
+import resolveConfig from "tailwindcss/resolveConfig";
+import tailwindConfig from "@sovryn/tailwindcss-config/index";
+import { MemoryRouter } from "react-router-dom";
 
 const config = resolveConfig(tailwindConfig);
 
 const screens = {
-  xs: '320px',
+  xs: "320px",
```
Comment (tiltom) [inline packages/ui/.storybook/preview.js]: This formatting is not correct, when I format this file I get single quotes that are correct.

**Review by tiltom — CHANGES_REQUESTED** [review]
Pls go through all of the files in this PR and make sure you have correct formatting.

File: packages/ui/package.json [hunk 7]
```diff
@@ -37,8 +37,12 @@
     "typescript": "^4.5.2"
   },
   "dependencies": {
+    "@fortawesome/fontawesome-svg-core": "^6.2.0",
```
Comment (tiltom) [inline packages/ui/package.json]: I added a new rule (after a discussion with @soulBit and @creed-victor) regarding packages versioning: https://sovryn.monday.com/docs/2733202752?blockId=aebe8f4b-5e4a-4ef6-8967-503ce7a1bb30 . Pls, adjust the whole file.

File: packages/ui/src/1_atoms/Button/Button.tsx [hunk 8]
```diff
@@ -1,32 +1,132 @@
...
+  xl = 'xl',
+}
+
+export enum ButtonStyle {
+  normal = 'normal',
+  inverted = 'inverted',
+  transparent = 'transparent',
+  frosted = 'frosted',
+  link = 'link',
+}
```
Comment (tiltom) [inline packages/ui/src/1_atoms/Button/Button.tsx]: Create a type file for the component (`Button.types.ts`) and extract it there.

File: packages/ui/src/1_atoms/Button/Button.tsx [hunk 9]
```diff
@@ -1,32 +1,132 @@
...
         className,
-      )}
-    >
-      {label}
-    </button>
+      ),
+    [loading, color, size, style, disabled, className],
   );
+
+  const onClickWhenAllowed = !disabled && !loading ? onClick : undefined;
```
Comment (tiltom) [inline packages/ui/src/1_atoms/Button/Button.tsx]:
```suggestion
  const onClickHandler = !disabled && !loading ? onClick : undefined;
```

Comment (tiltom) [reply packages/ui/src/1_atoms/Button/Button.tsx]: Use `useMemo` as well.

File: packages/ui/src/1_atoms/Button/Button.tsx [hunk 10]
```diff
@@ -1,32 +1,132 @@
...
+        onClick={onClickWhenAllowed}
+        data-action-id={dataActionId}
+      >
+        {text}
+      </button>
+    );
+  }
+};
+
+Button.defaultProps = {
```
Comment (tiltom) [inline packages/ui/src/1_atoms/Button/Button.tsx]: Do not use `defaultProps`, see https://sovryn.monday.com/docs/2733202752?blockId=8271db19-de51-4fd3-b6a0-88e3be3cb025 .

File: packages/ui/src/1_atoms/Icon/Icon.tsx [hunk 11]
```diff
@@ -0,0 +1,103 @@
...
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { IconProp, SizeProp } from '@fortawesome/fontawesome-svg-core';
+import classNames from 'classnames';
+import styles from './icon.module.css';
+import { IconSvgPaths } from './iconSvgPaths';
+import { IconType, ViewBoxSize } from './types';
+
+const STANDARD = 16;
+const INLINE = '1em';
+const SM = 'sm';
```
Comment (tiltom) [inline packages/ui/src/1_atoms/Icon/Icon.tsx]: Move them to `Icon.types.ts`.

File: packages/ui/src/1_atoms/Icon/Icon.tsx [hunk 12]
```diff
@@ -0,0 +1,103 @@
...
+  size = STANDARD,
+  inline,
+  className,
+}) => {
+  const isFaIcon = useMemo(() => {
+    if (icon && icon['prefix']) {
+      return true;
+    }
+    return false;
+  }, [icon]);
```
Comment (tiltom) [inline packages/ui/src/1_atoms/Icon/Icon.tsx]:
```suggestion
  const isFaIcon = useMemo(() => !!icon && !!icon['prefix'], [icon]);
```

File: packages/ui/src/1_atoms/Icon/Icon.tsx [hunk 13]
```diff
@@ -0,0 +1,103 @@
...
+    if (inline) {
+      return SM;
+    } else {
+      return size as SizeProp;
+    }
+  }, [inline, size]);
+
+  const inlineBlock = useMemo(() => {
+    return inline ? 'inline-block' : 'block';
+  }, [inline]);
```
Comment (tiltom) [inline packages/ui/src/1_atoms/Icon/Icon.tsx]:
```suggestion
  const inlineBlock = useMemo(() => inline ? 'inline-block' : 'block', [inline]);
```

File: packages/ui/src/2_molecules/Dialog/dialog.module.css [hunk 14]
```diff
@@ -0,0 +1,32 @@
+@import "../../../styles/index.css";
+
```
Comment (tiltom) [inline packages/ui/src/2_molecules/Dialog/dialog.module.css]: Rename the file to `Dialog.module.css`, make sure all of the components have proper file names.

File: packages/ui/src/2_molecules/Dialog/types.ts [hunk 15]
```diff
@@ -0,0 +1,35 @@
+import type { FunctionComponent } from 'react';
+
```
Comment (tiltom) [inline packages/ui/src/2_molecules/Dialog/types.ts]: Rename the file to `Dialog.types.ts`.

File: packages/ui/src/2_molecules/Dropdown/Dropdown.tsx [hunk 16]
```diff
@@ -0,0 +1,132 @@
...
+    if (!coords) {
+      return;
+    }
+    return getDropdownPositionStyles(coords, mode);
+  }, [coords, mode]);
+
+  const classNameComplete = useMemo(
+    () => classNames(styles.button, color, className, isOpen && styles.isOpen),
+    [color, className, isOpen],
+  );
```
Comment (tiltom) [inline packages/ui/src/2_molecules/Dropdown/Dropdown.tsx]:
```suggestion
  const classNameComplete = useMemo(
    () => classNames(styles.button, color, className, {
      [styles.isOpen] : isOpen
    }),
    [color, className, isOpen],
  );
```

File: packages/ui/src/2_molecules/Dropdown/Dropdown.tsx [hunk 17]
```diff
@@ -0,0 +1,132 @@
...
+
+  const useClickedOutside = useCallback(() => {
+    setOpen(false);
+    if (onClose) {
+      onClose();
+    }
+  }, [onClose]);
```
Comment (tiltom) [inline packages/ui/src/2_molecules/Dropdown/Dropdown.tsx]:
```suggestion
  const useClickedOutside = useCallback(() => {
    setOpen(false);
    onClose?.();
  }, [onClose]);
```

File: packages/ui/src/2_molecules/Dropdown/Dropdown.tsx [hunk 18]
```diff
@@ -0,0 +1,132 @@
...
+  useEffect(() => {
+    if (!isOpen) {
+      return;
+    }
+    const coords = getCoords();
+    setCoords(coords);
+
+    if (onOpen) {
+      onOpen();
+    }
```
Comment (tiltom) [inline packages/ui/src/2_molecules/Dropdown/Dropdown.tsx]:
```suggestion
    onOpen?.();
```

File: packages/ui/src/2_molecules/Dropdown/types.ts [hunk 19]
```diff
@@ -0,0 +1,21 @@
+export enum DropdownMode {
+  left = 'left',
+  right = 'right',
+  center = 'center',
+  sameWidth = 'sameWidth',
+}
+
```
Comment (tiltom) [inline packages/ui/src/2_molecules/Dropdown/types.ts]: Rename the file to `Dropdown.types.ts`.

File: packages/ui/src/2_molecules/Dropdown/utils.ts [hunk 20]
```diff
@@ -0,0 +1,56 @@
+import { DropdownCoords, DropdownMode } from './types';
+
```
Comment (tiltom) [inline packages/ui/src/2_molecules/Dropdown/utils.ts]: Rename the file to `Dropdown.utils.ts`. Make sure that every file in this PR is named according to https://sovryn.monday.com/docs/2733202752?blockId=275525d0-9985-4644-8023-61b8ead7d005 .

File: packages/ui/src/2_molecules/Menu/components/MenuItem/index.tsx [hunk 21]
```diff
@@ -0,0 +1,158 @@
...
+import classNames from 'classnames';
+import { Link, useLocation } from 'react-router-dom';
+import styles from './index.module.css';
+import { IconProp } from '@fortawesome/fontawesome-svg-core';
+import iconNewTab from '../../../../../assets/storybook/iconNewTab.svg';
+import { Icon } from '../../../../1_atoms';
+
+type MenuItemProps = {
+  className?: string;
+  //TODO: use IconComponents enum here instead, requires Icon Component to be done
```
Comment (tiltom) [inline packages/ui/src/2_molecules/Menu/components/MenuItem/index.tsx]: What about this comment? `Icon` component is done.

File: packages/ui/src/2_molecules/Menu/components/MenuSeparator/index.tsx [hunk 22]
```diff
@@ -0,0 +1,15 @@
...
+  className?: string;
+  text?: ReactNode;
+};
+
+export const MenuSeparator: React.FC<MenuSeparatorProps> = ({
+  className,
+  text,
+}) => {
+  return <li className={classNames(styles.host, className)}>{text}</li>;
+};
```
Comment (tiltom) [inline packages/ui/src/2_molecules/Menu/components/MenuSeparator/index.tsx]:
```suggestion
export const MenuSeparator: React.FC<MenuSeparatorProps> = ({
  className,
  text,
}) => <li className={classNames(styles.host, className)}>{text}</li>;
```

File: packages/ui/src/2_molecules/Overlay/Overlay.tsx [hunk 23]
```diff
@@ -0,0 +1,130 @@
...
+  [Align.left]: 'justify-start',
+  [Align.center]: 'justify-center',
+  [Align.right]: 'justify-end',
+};
+
+const AlignVerticalClassName: { [key in AlignVertical]: string } = {
+  [AlignVertical.top]: 'items-start',
+  [AlignVertical.center]: 'items-center',
+  [AlignVertical.bottom]: 'items-end',
+};
```
Comment (tiltom) [inline packages/ui/src/2_molecules/Overlay/Overlay.tsx]: Extract to the types file.

File: packages/ui/src/2_molecules/Overlay/Overlay.tsx [hunk 24]
```diff
@@ -0,0 +1,130 @@
...
+  background = OverlayBackground.dark75,
+  onBlur,
+  children,
+  portalClassName,
+}) => {
+  const onBlurHandler = useCallback(
+    (event: MouseEvent) => {
+      if (onBlur) {
+        onBlur(event);
+      }
```
Comment (tiltom) [inline packages/ui/src/2_molecules/Overlay/Overlay.tsx]:
```suggestion
        onBlur?.(event);
```

**Review by tiltom — CHANGES_REQUESTED** [review]

File: packages/ui/package.json [hunk 25]
```diff
@@ -37,8 +37,12 @@
     "typescript": "^4.5.2"
   },
   "dependencies": {
+    "@fortawesome/fontawesome-svg-core": "6.2.0",
+    "@fortawesome/free-solid-svg-icons": "6.2.0",
+    "@fortawesome/react-fontawesome": "0.2.0",
     "@sovryn/tailwindcss-config": "*",
-    "classnames": "^2.3.1"
+    "classnames": "^2.3.1",
```
Comment (tiltom) [inline packages/ui/package.json]:
```suggestion
    "classnames": "2.3.1",
```

Comment (tiltom) [reply packages/ui/package.json]: Remove `^` in the **whole** file (not only your changes).

File: packages/ui/src/2_molecules/Dropdown/Dropdown.tsx [hunk 26]
```diff
@@ -0,0 +1,128 @@
...
+import classNames from 'classnames';
+import styles from './Dropdown.module.css';
+import { useOnClickOutside } from '../../hooks/useOnClickOutside';
+
+import { DropdownColor, DropdownCoords, DropdownMode } from './Dropdown.types';
+import { getDropdownPositionStyles } from './Dropdown.utils';
+import { Nullable } from '../../types';
+import { Portal } from '../../1_atoms/Portal/Portal';
+
+type IDropdownProps = {
```
Comment (tiltom) [inline packages/ui/src/2_molecules/Dropdown/Dropdown.tsx]:
```suggestion
type DropdownProps = {
```

File: packages/ui/src/2_molecules/Menu/components/MenuItem/MenuItem.tsx [hunk 27]
```diff
@@ -0,0 +1,157 @@
...
+    event => {
+      if (disabled) {
+        event.preventDefault();
+        event.stopPropagation();
+      } else if (onClick) {
+        onClick(event);
+      }
+    },
+    [onClick, disabled],
+  );
```
Comment (tiltom) [inline packages/ui/src/2_molecules/Menu/components/MenuItem/MenuItem.tsx]:
```suggestion
  const onClickHandler = useCallback(
    event => {
      if (disabled) {
        event.preventDefault();
        event.stopPropagation();
      } else {
        onClick?.(event);
      }
    },
    [onClick, disabled],
  );
```

**Review by tiltom — CHANGES_REQUESTED** [review]

**Comment by pietro-maximoff** [comment]
@Rickk137 @tiltom I can't find any discussion about why we use CSS instead of SCSS in atom components or we planning to update them later?

**Comment by Rickk137** [comment]
> @Rickk137 @tiltom I can't find any discussion about why we use CSS instead of SCSS in atom components or we planning to update them later?

We have decide to stop using scss. @tiltom will update the code guideline

**Comment by tiltom** [comment]
> 

We decided not to use `Sass` in one of our spikes regarding the new dapp. It's not something I think we need to have in guidelines but I updated the existing guidelines so we don't mention `scss` anymore.

**Review by tiltom — APPROVED** [review]

**Review by pietro-maximoff — APPROVED** [review]
looks good to me, there was an issue with css file naming, discussed it with Rick, and fixed with `git config core.ignorecase true`

**Review by soulBit — APPROVED** [review]

**Pull request merged**